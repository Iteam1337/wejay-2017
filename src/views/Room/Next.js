import React from 'react'
import { gql, graphql } from 'react-apollo'

export const Next = ({ mutate, roomName }) => {
  const handleNext = () => {
    mutate({
      variables: {
        roomName
      }
    })
  }

  return <button onClick={handleNext}>Next</button>
}

const nextMutation = gql`
  mutation nextTrack($roomName: String!) {
    nextTrack(roomName: $roomName) {
      name
    }
  }
`

export default graphql(nextMutation)(Next)
