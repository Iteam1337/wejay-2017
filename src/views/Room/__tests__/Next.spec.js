import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Next } from '../Next'

describe('components/Next', () => {
  let component
  let mutate

  beforeEach(() => {
    mutate = jest.fn()

    component = shallow(<Next mutate={mutate} roomName="test" />)
  })

  it('can render', () => {
    expect(toJson(component)).toMatchSnapshot()
  })

  it('handles the mutation on click', () => {
    component.simulate('click')

    expect(mutate).toHaveBeenCalledWith({
      variables: {
        roomName: 'test'
      }
    })
  })
})
