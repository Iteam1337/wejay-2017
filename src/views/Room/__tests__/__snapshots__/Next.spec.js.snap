// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Next can render 1`] = `
<button
  onClick={[Function]}
>
  Next
</button>
`;
